File: s1/L1/python/palindrom.py
#!/usr/bin/env python


def main():
    pn = int(input("Podaj liczbę: "))
    n = pn
    base = int(input("Podaj podstawę: "))
    rem = 0
    pal = 0

    if base == 0:
        print("System liczbowy nie moze opierac sie na zerze")
        return
    if base == 1:
        print("Wszystko w systemie jedynkowym to palindrom")
        return
    if base < 0:
        print("System liczbowy nie moze byc ujemny")
        return
    if pn < 0:
        print("Liczba nie moze byc ujemna")
        return

    while n != 0:
        rem = n % base
        n //= base
        pal = base * pal + rem

    if pn == pal:
        print(f"{pn} jest palindromem")
    else:
        print(f"{pn} nie jest palindromem")


if __name__ == "__main__":
    main()

File: s1/L1/c/palindrom.c
#include <stdio.h>
#include <stdint.h>

int is_palindrome(uint32_t num, uint32_t base){
    uint32_t pal = 0;
    uint32_t rem = 0;
    uint32_t n = num;

    while(n != 0){
        rem = n % base;
        n /= base;

        pal = pal * base + rem;
    }

    return num == pal;
}

int main(){
    uint32_t num = 0;
    uint32_t base = 0;
    scanf("%u", &num);
    scanf("%u", &base);
    if(base == 0){
        printf("System liczbowy nie moze opierac sie na zerze\n");
        return 1;
    }
    if(base == 1){
        printf("Tak, wszystko w systemie 1 jest palindromem\n");
        return 1;
    }
    int pal = is_palindrome(num, base);
    printf("%u%s jest palindromem\n", num, pal == 0 ? " nie" : "");

    return 0;
}
